pkg/procsyms/procsyms.go

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package procsyms
+
+import (
+	"debug/elf"
+	"fmt"
+	"sync"
+
+	"github.com/cilium/tetragon/pkg/logger"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/prometheus/procfs"
+)
+
+var (
+	cache *lru.Cache[struct {
+		module string
+		offset uint64
+	}, string]
+	setCache sync.Once
+)
+
+// FnSym is a function location (function name, module path + offset)
+type FnSym struct {
+	Name   string
+	Module string
+	Offset uint64
+}
+
+// ToString returns a string representation of FnSym
+func (fsym *FnSym) ToString() string {
+	return fmt.Sprintf("%s (%s+0x%x)", fsym.Name, fsym.Module, fsym.Offset)
+}
+
+// GetFnSymbol -- returns the FnSym for a given address and PID
```

Comment:
It should be documented that this function is limited to native code only (C, C++, Go and Rust). The given approach will not work for interpreted languages (Java, Python, Perl, etc), as it can not resolve the internal state of the interpreter.